lisa/notifiers/log_file.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+from dataclasses import dataclass
+from typing import Any, List, Type, cast
+
+from dataclasses_json import dataclass_json
+
+from lisa import notifier, schema
+from lisa.environment import Environment
+from lisa.testsuite import TestResultMessage
+from lisa.util import constants, dump_file
+
+
+@dataclass_json()
+@dataclass
+class LogFileSchema(schema.TypedSchema):
+    log_level: str = logging.getLevelName(logging.DEBUG)
+
+
+class LogFile(notifier.Notifier):
+    """
+    It's a sample notifier, output subscribed message to a log file.
+    It can be used to troubleshooting what kind of message received.
+    """
+    
+
+    @classmethod
+    def type_name(cls) -> str:
+        return constants.NOTIFIER_LOG_FILE
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return LogFileSchema
+
+    def _received_message(self, message: notifier.MessageBase) -> None:
+        print("who do we code")
+        self._log.log(
+            getattr(logging, self._log_level),
+            f"log file received message [{message.type}]: {message}",
+        )
+        dump_file(self.log_file_path, message)
+
+    def _subscribed_message_type(self) -> List[Type[notifier.MessageBase]]:
+        return [TestResultMessage, notifier.TestRunMessage]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        runbook = cast(LogFileSchema, self.runbook)
+        self._log_level = runbook.log_level
+        self.log_file_path = environment.log_path / "log_file.txt"
```

Comment:
Im not sure how to get the environment for this line. Should I add it as an argument ?